extensions/cba_settings/cba_settings.cpp

```diff
@@ -0,0 +1,335 @@
+return
+
+#define RETURN(val)\
+strncpy_s(output, outputSize, result.c_str(), _TRUNCATE);\
+return (val)
+
+#define SYNTAX_ERROR_WRONG_PARAMS_SIZE 101
+#define SYNTAX_ERROR_WRONG_PARAMS_TYPE 102
+#define PARAMS_ERROR_TOO_MANY_ARGS 201
+#define EXECUTION_WARNING_TAKES_TOO_LONG 301
+
+#define TEMP_FILE "userconfig/temp"
+#define SETTINGS_FILE "userconfig/cba_settings.sqf"
+#define DELIM ":"
+#define MAX_CALLBACK_LENGTH 10000
+
+using namespace std;
+
+string version() {
+    return VERSION;
+}
+
+string clear() {
+    ofstream file;
+    file.open(SETTINGS_FILE, ios::out);
+    file.close();
+    return "true";
+}
+
+string load(int strlen) {
+    string result = "";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        if (file.good() || line.length() != 0) // Don't explode file by appending duplicate additional newlines.
+            result = result + trim(line) + "\n";
+    }
+
+    file.close();
+
+    if (strlen * MAX_CALLBACK_LENGTH > result.length())
+        return "";
+
+    result = result.substr(strlen * MAX_CALLBACK_LENGTH, MAX_CALLBACK_LENGTH);
+    return result;
+}
+
+string parse(int strlen) {
+    string result = "[";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+
+        // Ignore commented out lines.
+        if (line.substr(0, 2) == "//")
+            continue;
+
+        // Count force tags.
+        int force = 0;
+        while (line.substr(0, 6) == "force ") {
+            force++;
+            line = line.substr(6);
+            line = trim_left(line);
+        }
+
+        // Get setting.
+        size_t i = line.find("=");
+        if (i == string::npos) // Ignore ill-formed line.
+            continue;
+
+        string item = line.substr(0, i);
+        item = trim_right(item);
+
+        // Get value.
+        string value = line.substr(i + 1);
+        value = trim(value);
+
+        while (value.back() == ';')
+            value.pop_back();
+
+        value = trim_right(value);
+
+        result = result + "[\"" + item + "\"," + value + "," + to_string(force) + "],";
+    }
+
+    file.close();
+
+    result.pop_back();
+    result.push_back(']');
+
+    if (strlen * MAX_CALLBACK_LENGTH > result.length())
+        return "";
+
+    result = result.substr(strlen * MAX_CALLBACK_LENGTH, MAX_CALLBACK_LENGTH);
+    return result;
+}
+
+string read(string setting) {
+    string result = "[\"\",false,-1]";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+
+        // Ignore commented out lines.
+        if (line.substr(0, 2) == "//")
+            continue;
+
+        // Count force tags.
+        int force = 0;
+        while (line.substr(0, 6) == "force ") {
```

Comment:
you are copying the substring on every iteration here. If you use the string_view trim, it will only move a pointer around